openhtf/io/output/__init__.py

```diff
@@ -1,26 +1,41 @@
-# Copyright 2014 Google Inc. All Rights Reserved.
+"""Output module."""
 
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+import base64
+import os
+import tempfile
+from openhtf.util import data
 
-#     http://www.apache.org/licenses/LICENSE-2.0
 
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+class OutputToFile(object):
+  """Class for formatting filename and writing to file."""
 
-"""Openhtf output modules."""
+  def __init__(self, filename_pattern=None):
+    self.filename_pattern = filename_pattern
 
-import openhtf.io.output.json_factory
-import openhtf.io.output.mfg_inspector
+  def __call__(self, test_record):
+    test_record_dict = self.ConvertToDict(test_record)
+    write_data = self.GetWriteData(test_record)
 
-# pylint: disable=invalid-name
+    if isinstance(self.filename_pattern, basestring):
+      filename = self.Format(test_record_dict)
+      self.Save(filename, write_data)
+      return filename
+    self.filename_pattern.write(write_data)
 
-# Classes used for output
-OutputToJSON = json_factory.OutputToJSON
-UploadToMfgInspector = mfg_inspector.UploadToMfgInspector
-OutputToTestRunProto = mfg_inspector.OutputToTestRunProto
-UploadOrOutput = mfg_inspector.UploadOrOutput
+  def ConvertToDict(self, test_record):
+    return data.ConvertToBaseTypes(test_record)
+
+  def GetWriteData(self, test_record):
+    return str(self.ConvertToDict(test_record))
+
+  def Format(self, test_record_dict):
+    if '{' in self.filename_pattern:
+      return self.filename_pattern.format(**test_record_dict)
+    return self.filename_pattern % test_record_dict
```

Comment:
let's raise a `ValueError('%-style filename patterns deprecated, use .format() syntax')` in this case.